backend/PostProcessing/PostProcessingClasses/LinearInterpolation.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#ImmediateArithmeticOperation.py
+#-------------------------------
+# Created By : Matthew Kastl
+#-------------------------------
+""" The post processing in this file preforms an linear interpolation of a column.
+ """ 
+#-------------------------------
+# 
+#
+#Imports
+from PostProcessing.IPostProcessing import IPostProcessing
+from pandas import DataFrame, date_range
+import numpy as np
+from datetime import timedelta
+
+class LinearInterpolation(IPostProcessing):
```

Comment:
My main question is whether this is meant to replace the post-processing class used for interpolating between measurements and predictions. If so, how will this be implemented ? Say we have three models that predict at 12-hour, 24-hour, and 48-hour intervals. You would need to take one prediction from each model, along with the last available measurement from the measured data, and create a data frame containing these data points. Then, you would generate null data points between the values and pass this data frame into this post-processing class. When exactly would this process occur?


A visual eaxample of what I mean:

| Date               | Value            |
|--------------------|------------------|
| 2024-11-20 00:00   | measurement      |
| 2024-11-20 01:00   | null             |
| 2024-11-20 02:00   | null             |
| 2024-11-20 03:00   | null             |
| 2024-11-20 04:00   | null             |
| 2024-11-20 05:00   | null             |
| 2024-11-20 06:00   | null             |
| 2024-11-20 07:00   | null             |
| 2024-11-20 08:00   | null             |
| 2024-11-20 09:00   | null             |
| 2024-11-20 10:00   | null             |
| 2024-11-20 11:00   | null             |
| 2024-11-20 12:00   | 12 hr prediction |
| 2024-11-20 13:00   | null             |
| 2024-11-20 14:00   | null             |
| 2024-11-20 15:00   | null             |
| 2024-11-20 16:00   | null             |
| 2024-11-20 17:00   | null             |
| 2024-11-20 18:00   | null             |
| 2024-11-20 19:00   | null             |
| 2024-11-20 20:00   | null             |
| 2024-11-20 21:00   | null             |
| 2024-11-20 22:00   | null             |
| 2024-11-20 23:00   | null             |
| 2024-11-21 00:00   | 24 hr prediction |
| 2024-11-21 01:00   | null             |
| 2024-11-21 02:00   | null             |
| 2024-11-21 03:00   | null             |
| 2024-11-21 04:00   | null             |
| 2024-11-21 05:00   | null             |
| 2024-11-21 06:00   | null             |
| 2024-11-21 07:00   | null             |
| 2024-11-21 08:00   | null             |
| 2024-11-21 09:00   | null             |
| 2024-11-21 10:00   | null             |
| 2024-11-21 11:00   | null             |
| 2024-11-21 12:00   | null             |
| 2024-11-21 13:00   | null             |
| 2024-11-21 14:00   | null             |
| 2024-11-21 15:00   | null             |
| 2024-11-21 16:00   | null             |
| 2024-11-21 17:00   | null             |
| 2024-11-21 18:00   | null             |
| 2024-11-21 19:00   | null             |
| 2024-11-21 20:00   | null             |
| 2024-11-21 21:00   | null             |
| 2024-11-21 22:00   | null             |
| 2024-11-21 23:00   | null             |
| 2024-11-22 00:00   | 48 hr prediction |